src/storage/background_process.cpp

```diff
@@ -87,4 +102,17 @@ void BGTaskProcessor::Process() {
     }
 }
 
+void BGTaskProcessor::UpdateCheckpointState(TxnTimeStamp max_commit_ts, i64 wal_size) {
```

Comment:
Maybe the checkpoint state should be stored in catalog if we must save the information.